document t/test.pl

tonycoz opened the pull request:
I got sick of reverse engineering run_multiple_progs() tests and runperl() arguments each time I needed to use them.

There's some documentation in comments inline but it's pretty variable and less accessible than pod.

Since (in theory anyway) we want test.pl to exercise as little of perl as possible, the POD doesn't belong in test.pl itself.

So I've put this in t/test_pl.pod since it's not really end user documentation that would belong in pod/.

Suggestions welcome for a better location.

* This set of changes does not require a perldelta entry.

tonycoz committed:
document t/test.pl

I got sick of reverse engineering run_multiple_progs() tests and
runperl() arguments each time I needed to use them.

There's some documentation in comments inline but it's pretty variable
and less accessible than pod.

Since (in theory anyway) we want test.pl to exercise as little of perl
as possible, the POD doesn't belong in test.pl itself.

So I've put this in t/test_pl.pod since it's not really end user
documentation that would belong in pod/.

jkeenan reviewed (commented):
There is a lot to review here.  The sections up through `Test::More differences` look good to me.  I suggest getting different people to review the various remaining sections:
```
204:=head1 Extended test functions
438:=head1 Utility functions
661:=head1 Runperl Arguments
736:=head1 run_multiple_progs() tests
```

jkeenan commented on t/test_pl.pod:
```diff
@@ -0,0 +1,855 @@
+
+=head1 Extended test functions
+
+These test some condition and produce TAP for a test.
```
At this point in my life, I know what `TAP` is when I see it ... but I had to stop and think what the acronym stands for.  Perhaps a link to someplace that defines it would help.

tonycoz replied:
I was mostly going for "produces a counted test" but I'm not sure that's better

jkeenan commented on t/test_pl.pod:
```diff
@@ -658,6 +659,32 @@ Returns true if the test appears to be running in a container.
 
 =back
 
+There's also some variables which detect useful configuration:
```
Grammar: should be "There are also ..."

khwilliamson commented:
I think we should just merge this.  It lgtm, but there is a lot of stuff here that I didn't know about.  If something comes up where this isn't quite right, what harm will have been done.  We would just fix it.  End-users would not be affected.

tonycoz committed:
podcheck t/test_pl.pod

podcheck normally excludes t/ but I do want t/test_pl.pod checked.

Updating is_pod_file() to allow only this file seems more complex than
just adding t/test_pl.pod afterwards, so just add it, just as
t/porting/podcheck.t is added.

tonycoz commented:
added it to podcheck and fixed some issues (`C</dev/null>` to `F</dev/null>` and some verbatim overflows.
